src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2218,6 +2218,31 @@ public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, Syn
             // is/as are valid after expressions.
             if (token.IsLastTokenOfNode<ExpressionSyntax>())
             {
+                var expression = token.GetAncestor<ExpressionSyntax>();
+                var symbols = semanticModel.GetSymbolInfo(expression).GetBestOrAllSymbols();
+                var methodSymbols = symbols.OfType<IMethodSymbol>();
+                if (methodSymbols.Any())
```

Comment:
❓ What happens if `symbols` contains an `IMethodSymbol` and an `IPropertySymbol` (is this possible in ambiguous reference scenarios?)? Concerned that the current approach would ignore the property symbol even though the keywords would be valid after it.